### PR TITLE
DT-969: Show server errors when creating a draft DAR

### PR DESCRIPTION
**Pull request by rushtong**
### Addresses
https://broadworkbench.atlassian.net/browse/DT-969

### Summary
If there are server errors when submitting a draft dar, they are swallowed by the UI. This PR shows them in the two cases where one can apply for access to a dataset(s).

### Data Library View
https://github.com/user-attachments/assets/43b8378b-ea85-4d50-9d59-7f5c5a317e2f

### Dataset View
https://github.com/user-attachments/assets/e3072377-101c-4594-9840-223aac56df3e

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by snf2ye — APPROVED** [review]
LGTM - thanks for adding the videos!

File: src/pages/DatasetStatistics.jsx [hunk 1]
```diff
@@ -20,8 +20,18 @@ export default function DatasetStatistics(props) {
...
-    const darDraft = await DAR.postDarDraft({ datasetId: [datasetId]  });
-    history.push(`/dar_application/${darDraft.referenceId}`);
+    try {
+      const draftResponse = await DAR.postDarDraft({ datasetId: [datasetId]  });
+      if (draftResponse.referenceId) {
+        history.push(`/dar_application/${draftResponse.referenceId}`);
+      } else if (draftResponse.message) {
+        Notifications.showError({ text: draftResponse.message + ' Please contact customer support for help.' });
+      } else {
+        Notifications.showError({ text: 'Error: Unable to create a Draft Data Access Request' });
```
Comment (fboulnois) [inline src/pages/DatasetStatistics.jsx:30]: Possible improvement, `throw` here so it bubbles to the `catch` statement.

File: src/components/data_search/DatasetSearchTable.jsx [hunk 2]
```diff
@@ -187,8 +187,18 @@ export const DatasetSearchTable = (props) => {
...
-    const darDraft = await DAR.postDarDraft({ datasetId: selected });
-    history.push(`/dar_application/${darDraft.referenceId}`);
+    try {
+      const draftResponse = await DAR.postDarDraft({ datasetId: selected  });
+      if (draftResponse.referenceId) {
+        history.push(`/dar_application/${draftResponse.referenceId}`);
+      } else if (draftResponse.message) {
+        Notifications.showError({ text: draftResponse.message + ' Please contact customer support for help.' });
+      } else {
+        Notifications.showError({ text: 'Error: Unable to create a Draft Data Access Request' });
```
Comment (fboulnois) [inline src/components/data_search/DatasetSearchTable.jsx:197]: Ditto here.

**Review by fboulnois — APPROVED** [review]
Small suggested improvement:

File: src/pages/DatasetStatistics.jsx (hunk 1 above)
Comment (rushtong) [reply src/pages/DatasetStatistics.jsx:30]: I am seeing [this warning](https://www.jetbrains.com.cn/en-us/help/inspectopedia/ExceptionCaughtLocallyJS.html) with this suggestion. Another approach I would like to take is to extract the Apply functionality to a separate component - if we do tackle that, maybe we can improve the control flow there.

**Pull request merged**